Assets/Mona/TemplateHelper/TemplateHelper.cs

```diff
@@ -0,0 +1,246 @@
+                {
+                    templateVersion = new TemplateVersion();
+                    JsonUtility.FromJsonOverwrite(webReq.downloadHandler.text, templateVersion);
+                    string newestVersion = templateVersion.records[0].fields.version;
+                    if (!(versionCompare(TemplateInfo.Version, newestVersion) < 0))
+                    {
+                        Update = "None";
+                        TemplateCheck.UpdateAvaliable = false;
+                    }   
+                    else
+                    {
+                        Update = templateVersion.records[0].fields.file[0].url;
+                        FileName = templateVersion.records[0].fields.file[0].filename;
+                        TemplateCheck.UpdateAvaliable = true;
+                        AssetDatabase.Refresh();
+                        Debug.LogWarning("Newer version of the Mona StarterSpace Template is avaliable (" + newestVersion + "). Visit the template helper utility to easily download updates.");
+                    }
+                }
+            }
+        }
+        public bool HasCorrectUVersion()
+        {
+            if (Application.unityVersion.Contains(TemplateInfo.UnityVer))
+            {
+                return true;
+            }
+            return false;
+        }
+        public bool HasWebGLModule()
+        {
+            var moduleManager = System.Type.GetType("UnityEditor.Modules.ModuleManager,UnityEditor.dll");
+            var isPlatformSupportLoaded = moduleManager.GetMethod("IsPlatformSupportLoaded", System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic);
+            var getTargetStringFromBuildTarget = moduleManager.GetMethod("GetTargetStringFromBuildTarget", System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic);
+            if ((bool)isPlatformSupportLoaded.Invoke(null, new object[] {(string)getTargetStringFromBuildTarget.Invoke(null, new object[] {BuildTarget.WebGL})}))
+            {
+                return true;
+            }
+            return false;
+        }
+        private int versionCompare(string v1, string v2)
+        {
+            int vnum1 = 0, vnum2 = 0;
+            for (int i = 0, j = 0; (i < v1.Length || j < v2.Length);)
+            {
+                while (i < v1.Length && v1[i] != '.')
+                {
+                    vnum1 = vnum1 * 10 + (v1[i] - '0');
+                    i++;
+                }
+                while (j < v2.Length && v2[j] != '.')
+                {
+                    vnum2 = vnum2 * 10 + (v2[j] - '0');
+                    j++;
+                }
+                if (vnum1 > vnum2)
+                {
+                    return 1;
+                }
+                if (vnum2 > vnum1)
+                {
+                    return -1;
+                }
+                vnum1 = vnum2 = 0;
+                i++;
+                j++;
+            }
+            return 0;
+        }
```

Comment:
This section needs comments on what's going on.  I'd also add a method level comment with sample inputs to make it clearer.